charts/turing/templates/turing-deployment.yaml

```diff
@@ -24,7 +24,9 @@ spec:
 {{ toYaml .Values.deployment.labels | indent 8 -}}
 {{- end }}
     spec:
+      {{- if (or .Values.serviceAccount.create .Values.serviceAccount.name) }}
       serviceAccountName: {{ template "turing.serviceAccountName" . }}
+      {{- end }}
```

Comment:
@krithika369 I made this change and the reason for which is similar to what you had originally pointed out for the kaniko service account i.e. we shouldn't be inserting a service account name if we are not expecting one to be created or if it's name is not specified anywhere